homeassistant/components/samsungtv/__init__.py

```diff
@@ -1 +1,49 @@
 """The Samsung TV integration."""
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT, CONF_TIMEOUT
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_NAME, DEFAULT_TIMEOUT, DOMAIN
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            [
+                vol.Schema(
+                    {
+                        vol.Required(CONF_HOST): cv.string,
+                        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                        vol.Optional(CONF_PORT): cv.port,
+                        vol.Optional(CONF_MAC): cv.string,
+                        vol.Optional(
+                            CONF_TIMEOUT, default=DEFAULT_TIMEOUT
+                        ): cv.positive_int,
+                    }
+                )
+            ]
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Samsung TV integration."""
+    if DOMAIN in config:
+        for entry_config in config[DOMAIN]:
+            await hass.config_entries.flow.async_init(
```

Comment:
Probably add a check here if the config is already matching a config entry.

Reply:
This check is already implemented in user config flow step.

Reply:
We can't await this. That's why you get the stack trace. We need to create a task. See other integrations with config flow for examples.